### PR TITLE
feat: convert to typescript

### PR DESCRIPTION
**Pull request by gustavkj**
- Converts `aoc-loader` to TypeScript.
- Adds some tooling for better DX.
- Replaces [`request-promise-native`](https://github.com/request/request-promise-native) and [`request`](https://github.com/request/request) (that have been deprecated in February 2020) with [`got`](https://github.com/sindresorhus/got).

### CODE REVIEW

**Review by lindell — APPROVED** [review]
LGTM 🚀 
Edit: Looked a little closer, there are some comments :)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
```
Comment (lindell) [inline .editorconfig]: I'm not a fan of 2 spaces, but I guess it's the JS standard now.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+max_line_length = 100
```
Comment (lindell) [inline .editorconfig]: Maybe 120 length? :)

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (lindell) [inline .vscode/extensions.json]: Editor specific files should be kept out of the repo :)

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (lindell) [inline .vscode/settings.json]: Editor specific files should be kept out of the repo :)

File: .editorconfig (hunk 1 above)
Comment (gustavkj) [reply .editorconfig]: I've changed it to 4 spaces now. 🙂

File: .editorconfig (hunk 2 above)
Comment (gustavkj) [reply .editorconfig]: Sure, I've updated it now

File: .vscode/extensions.json (hunk 3 above)
Comment (gustavkj) [reply .vscode/extensions.json]: While I would agree with you in the past, I think there are advantages in including settings in bigger project where many people work in the same editor and need to use the same tooling. But this is not a big project, so I'm removing them.

**Review by lindell — APPROVED** [review]
💯

**Pull request merged**